clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt

```diff
@@ -185,6 +183,21 @@ interface FossIdRestService {
         }
     }
 
+    /**
+     * An interceptor to set the timeout of the requests based on the call headers. If no timeout header is present,
+     * default timeout of the client is used.
+     */
+    object TimeoutInterceptor : Interceptor {
+        override fun intercept(chain: Interceptor.Chain): okhttp3.Response {
+            val request = chain.request()
+            val newReadTimeout = request.header(READ_TIMEOUT_HEADER)
+
+            val readTimeout = newReadTimeout?.takeIf { it.isNotEmpty() }?.toInt() ?: chain.readTimeoutMillis()
```

Comment:
Nit: Could be shortened to `newReadTimeout?.toIntOrNull() ?: chain.readTimeoutMillis()`.